### ISSUE TITLE
1.43.1 crashing and failing to read contacts

### ISSUE THREAD

**Issue by anaskaejdar** [deltalab-org/deltalab-android#17]
- Android version: 14
- Device: Samsung GS22 Ultra
- Delta Chat version: 1.43.1
- Expected behavior: Read contacts and set display names to match what they are in contacts.
- Actual behavior: display name remains just the contact's email address
- Steps to reproduce the problem:
  - upgrade to 1.43.1
  - Press the `+` button to refresh contacts
  - I don't know what else... I guess just check the names
  - Also, try going into the settings and click “Advanced.” Watch the app crash.
  - I should mention, both these things worked perfectly on 1.43.0
- Screenshots:
- Logs:
```
DeltaLab 1.43.1 Crash Report

Android 14:
java.lang.NullPointerException: Attempt to write to field 'b1.m androidx.preference.Preference.r' on a null object reference in method 'void xd.d.T(android.os.Bundle)'
	at xd.d.T(SourceFile:105)
	at androidx.fragment.app.s0.e(SourceFile:72)
	at androidx.fragment.app.s0.k(SourceFile:141)
	at androidx.fragment.app.n0.y(SourceFile:11)
	at androidx.fragment.app.n0.Q(SourceFile:82)
	at androidx.fragment.app.n0.x(SourceFile:74)
	at androidx.fragment.app.x.run(SourceFile:12)
	at android.os.Handler.handleCallback(Handler.java:958)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:230)
	at android.os.Looper.loop(Looper.java:319)
	at android.app.ActivityThread.main(ActivityThread.java:8893)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
```

**Comment by adbenitez** [comment]
hello @anaskaejdar thanks for filling the report, the issue was already fixed, there is a new 1.43.2 from 5 hours ago, you can download it already from the release section at:
https://github.com/deltalab-org/deltalab-android/releases

it might take some time for the update to appear in IzzyOnDroid

**Comment by adbenitez** [comment]
> * Expected behavior: Read contacts and set display names to match what they are in contacts.

this is unrelated, the app now doesn't automatically load contacts from your phone address book or request the "read contacts" permission, instead you have to enable contact-loading in the Privacy settings of the app, then the contacts will be loaded, also notice that if you load contacts this way they will not have encryption (unless the contact writes to you from delta) and unencrypted messages are not supported by chatmail instances (like the new "create account" option in the welcome screen

**Comment by anaskaejdar** [comment]
@adbenitez 

> > * Expected behavior: Read contacts and set display names to match what they are in contacts.
> 
> this is unrelated, the app now doesn't automatically load contacts from your phone address book or request the "read contacts" permission, instead you have to enable contact-loading in the Privacy settings of the app, then the contacts will be loaded, also notice that if you load contacts this way they will not have encryption (unless the contact writes to you from delta) and unencrypted messages are not supported by chatmail instances (like the new "create account" option in the welcome screen

It is actually related. I'm sorry, I didn't speak clearly. I did check that box. Otherwise, it wouldn't have been working properly on v1.43.0

**Comment by anaskaejdar** [comment]
> It is actually related. I'm sorry, I didn't speak clearly. I did check that box. Otherwise, it wouldn't have been working properly on v1.43.0

but nonetheless, it seems to be fixed in 1.43.2
Thanks for all you do brother
